File: RafJira/app/src/main/java/com/example/view/fragments/LoginFragment.java
package com.example.view.fragments;

import androidx.fragment.app.FragmentTransaction;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.example.constants.Constants;
import com.example.raf_jira.R;
import com.example.raf_jira.databinding.FragmentLoginBinding;

public class LoginFragment extends Fragment {

    private FragmentLoginBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentLoginBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init();
    }

    private void init(){
        binding.loginButton.setOnClickListener(view -> {
            String username = binding.usernameEt.getText().toString().trim();
            String email = binding.emailEt.getText().toString().trim();
            String password = binding.passwordEt.getText().toString().trim();

            if(username.isEmpty() || email.isEmpty() || password.isEmpty()){
                Toast.makeText(getActivity(), "Invalid inputs", Toast.LENGTH_SHORT).show();
                return;
            }
            if( !username.startsWith("admin") && !password.equals(Constants.USER_PASSWORD) ||
                    username.startsWith("admin") && !password.equals(Constants.ADMIN_PASSWORD) ){
                Toast.makeText(getActivity(), "Invalid username or password", Toast.LENGTH_SHORT).show();
                return;
            }

            SharedPreferences sharedPreferences = getActivity().getSharedPreferences(Constants.SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE);

            sharedPreferences
                    .edit()
                    .putBoolean(Constants.IS_LOGGED_KEY, true)
                    .putString(Constants.USERNAME_KEY, username)
                    .putString(Constants.EMAIL_KEY, email)
                    .putString(Constants.PASSWORD_KEY, password)
                    .apply();

            FragmentTransaction transaction = getParentFragmentManager().beginTransaction();
            transaction.addToBackStack(null);
            transaction.replace(R.id.fragment_container_view_tag, new MainFragment());
            transaction.commit();
        });
    }

}

File: Dnevnjak/app/src/main/java/com/example/dnevnjak/presentation/fragments/LoginFragment.kt
package com.example.dnevnjak.presentation.fragments

import android.content.Context
import android.os.Bundle
import android.provider.SyncStateContract
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.ui.platform.ComposeView
import androidx.fragment.app.Fragment
import com.example.dnevnjak.R
import com.example.dnevnjak.presentation.composable.LoginPage
import com.example.dnevnjak.utilities.Constants.Companion.EMAIL_KEY
import com.example.dnevnjak.utilities.Constants.Companion.MAIN_FRAGMENT_TAG
import com.example.dnevnjak.utilities.Constants.Companion.SHARED_PREFERENCES_PATH
import com.example.dnevnjak.utilities.Constants.Companion.USERNAME_KEY

class LoginFragment: Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        return ComposeView(requireContext()).apply {
            setContent {
                LoginPage(
                    onSuccess = { username, email -> onLoginSuccess(username, email) }
                )
            }
        }
    }

    private fun onLoginSuccess(username: String, email: String){

        Log.e("TAG", "$username $email")

        val sharedPreferences = requireActivity().getSharedPreferences(SHARED_PREFERENCES_PATH, Context.MODE_PRIVATE)
        sharedPreferences.edit().apply {
            putString(EMAIL_KEY,  email)
            putString(USERNAME_KEY, username)
            apply()
        }

        val transaction = activity?.supportFragmentManager?.beginTransaction()
        transaction?.replace(R.id.activity_main_fragment_container, MainFragment(), MAIN_FRAGMENT_TAG)
        transaction?.commit()
    }
}

